### ISSUE TITLE
[FEATURE]: Translate Rich Text Blocks

### ISSUE THREAD

**Issue by jsadoski-rockhall** [Fekide/strapi-plugin-translate#401]
**Is your feature request related to a problem? Please describe.**
When using the new **Rich Text (Blocks)** field, the content is not translated.

**Describe the solution you'd like**
Rich Text (Blocks) fields should be translated to the target locale.

**Describe alternatives you've considered**
For now, this field type is unusable if automated translation is required. Instead, we must continue to use Rich Text (Markdown) text fields.

**Additional context**
We use Deepl to translate our content. Given the structure of Blocks objects in the Strapi API, it may be difficult to send to the translation API and get back a meaningful response that respects grammar, etc.

Sorry if this issue has already been raised, but I didn't find any previous GH issues.

**Comment by boardmain** [comment]
Any updates?

**Comment by sargreal** [comment]
Hi @jsadoski-rockhall and @boardmain,

Thank you for your interest in this plugin and the feature proposal!

This is definitely an important feature that should be implemented in this plugin. However currently, we as maintainers have more pressing issues on the internal side and are also working voluntarily. If you require this feature, we would be very happy if you would try to implement it yourself and create a pull request. We will give you feedback on problems you may encounter.

From my understanding, the easiest way to implement this feature would be to create a rendering from the blocks syntax to html or xml, then translate that on the provider, and then convert it back to the blocks syntax.
There already exists an [html renderer for the blocks syntax](https://www.npmjs.com/package/blocks-html-renderer), so that may be used and only the translation back to the blocks syntax is required.

An issue that may definitely arise is that the rendering to html is (at least in the referenced html renderer) a lossy conversion, as not all properties are used in the html. I am not quite sure how this can be handled yet.

Another problem may arise when the document is too large to be translated in one request. Currently, there is also no good handling for splitting html documents, but this problem should definitely considered.

**Comment by boardmain** [comment]
@sargreal 
hi im doing some test
adding a custom provider ( deepl modified )

but if i try to add for example to the strapi-plugin-translate configurator inside plugins.js

translatedFieldTypes: [
        'string',
        **{ type: 'blocks', format: 'jsonb' },**
        { type: 'text', format: 'plain' },
        { type: 'richtext', format: 'markdown' },
        'component',
        'dynamiczone',
      ],

it is ignored by the plugin and i can't Handle this type in the custom provider
it is ignored into getTranslateFields

in the "getFieldTypeConfig(type) function"
when it look for the block type it never continue

~~~
function getFieldTypeConfig(type) {
  const { translatedFieldTypes } = strapi.config.get('plugin.translate')

	console.log('TYPE: ->', type);
	console.log(translatedFieldTypes);
  return translatedFieldTypes.find((t) => t === type || t.type === type)
}
~~~

i see:

~~~
TYPE: -> blocks
[
  { type: 'string', format: 'plain' },
  { type: 'text', format: 'plain' },
  { type: 'richtext', format: 'markdown' },
  'component',
  'dynamiczone'
]
~~~

can you help to modify the response of  strapi.config.get('plugin.translate') ? if it isn't handled by the plugins.js config gile

**Comment by sargreal** [comment]
@boardmain 
```js
  translatedFieldTypes: [
    'string',
    { type: 'blocks', format: 'jsonb' },
    { type: 'text', format: 'plain' },
    { type: 'richtext', format: 'markdown' },
    'component',
    'dynamiczone',
  ],
```

So first of all, there seems to be something wrong with your setup, since I get an expected Error `Error regarding translate config: unhandled format jsonb for translated field blocks` when using your config. If that error isn't thrown and you have not customized the config parsing, the config you are providing is not considered at all.

I believe the format parsing should completely be implemented on the plugin side, as the providers should only provide the translation function. The execution of the format conversion is only done on the provider side, since we cannot be sure which formats are supported by a provider (although that could also be stated in the provider's declaration as well. That would be a breaking change however).


Anyways, I have tried today a little bit at implementing this feature in a rudimentary state and will open a PR in the next days where you can have a look and maybe build from there.

**Comment by boardmain** [comment]
@sargreal yes i had an error in my plugins.js config file, now i have correctly the "Error regarding translate config"

**Comment by boardmain** [comment]
While this. update will be fixed, if anyone need it
i published this package 
strapi-provider-translate-deepljsonb 
it take all text from paragraph, translate it with deepl and writeback to object 

simply install it with strapi-provider-translate-deepljsonb  and replace deepl with deepljson i the plugin.js config

it is not perfect, but for now is working

**Comment by kristiczk** [comment]
> deepljsonb

Hello. I have tried your solution, but nothing changes :(
